x/evm/migrations/v4/types/params.go

```diff
@@ -0,0 +1,151 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/params"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/evmos/ethermint/types"
+)
+
+var _ paramtypes.ParamSet = &Params{}
```

Comment:
let's use the `LegacyParams` type that you introduce in the interfaces

```suggestion
var _ evm.LegacyParams = &Params{}
```